src/plugins/gmx/gmxplugin.cpp

```diff
@@ -0,0 +1,185 @@
+            stream.writeAttribute("scaleY", QString::number(1));
+
+            stream.writeEndElement();
+        }
+
+        if (file.error() != QFile::NoError) {
+            mError = file.errorString();
+            return false;
+        }
+
+    }
+
+    stream.writeEndElement();
+
+
+    stream.writeStartElement("tiles");
+
+
+
+    uint tileId = 0u;
+    int currentLayer = map->layers().size();
+
+    foreach (const Layer *layer, map->layers()) {
+
+        if (layer->layerType() != Layer::TileLayerType) continue;
+
+        const TileLayer *tileLayer = static_cast<const TileLayer*>(layer);
+        for (int y = 0; y < tileLayer->height(); ++y) {
+            for (int x = 0; x < tileLayer->width(); ++x) {
+                const Cell &cell = tileLayer->cellAt(x, y);
+                const Tile *tile = cell.tile;
+
+                if(tile) {
+                    stream.writeStartElement("tile");
+
+                    stream.writeAttribute("bgName", tile->tileset()->name());
+                    stream.writeAttribute("x", QString::number((int)(x * map->tileWidth())));
+                    stream.writeAttribute("y", QString::number((int)(y * map->tileHeight())));
+                    stream.writeAttribute("w", QString::number(map->tileWidth()));
+                    stream.writeAttribute("h", QString::number(map->tileHeight()));
+
+                    stream.writeAttribute("xo", QString::number(tile->id() % tile->tileset()->columnCount() * tile->tileset()->tileWidth()));
+
+                    stream.writeAttribute("yo", QString::number((int)(tile->id() / tile->tileset()->columnCount()) * tile->tileset()->tileWidth()));
+
+                    stream.writeAttribute("depth", QString::number(layer->hasProperty(QLatin1String("Depth")) ? layer->property(QLatin1String("Depth")).toInt() : currentLayer));
+                    stream.writeAttribute("id", QString::number(++tileId));
+                    stream.writeAttribute("scaleX", QString::number(1));
+                    stream.writeAttribute("scaleY", QString::number(1));
+
+                    stream.writeEndElement();
+                }
+            }
+        }
+
+        currentLayer--;
+
+        if (file.error() != QFile::NoError) {
+            mError = file.errorString();
+            return false;
+        }
+    }
+
+    stream.writeEndElement();
+    stream.writeEndDocument();
+
+    file.close();
+    return true;
+}
+
+QString GmxPlugin::errorString() const
+{
+    return mError;
+}
+
+QStringList GmxPlugin::outputFiles(const Tiled::Map *map, const QString &fileName) const
+{
+    QStringList result;
+
+    QFileInfo fileInfo(fileName);
+    const QString base = fileInfo.completeBaseName();
+    const QString path = fileInfo.path();
+
+    const QString completeName = base + QLatin1String(".room.gmx");
+    const QString completePath = QDir(path).filePath(completeName);
+
+    result.append(completePath);
+
+    if (result.size() == 1)
+        result[0] = fileName;
```

Comment:
What are you trying to achieve here? This function is effectively always just returning `fileName`, which is what the default implementation already does.